unstructured_inference/models/unstructuredmodel.py

```diff
@@ -55,6 +63,116 @@ def __call__(self, x: Image) -> List[LayoutElement]:
+        or merges or splits them depending on the iom score for both regions"""
+        intersections_mtx = intersections(*elements)
+
+        for i, row in enumerate(intersections_mtx):
+            first = elements[i]
+            if first:
+                # We get only the elements which intersected
+                indices_to_check = np.where(row)[0]
+                # Delete the first element, since it will always intersect with itself
+                indices_to_check = indices_to_check[indices_to_check != i]
+                if len(indices_to_check) == 0:
+                    continue
+                if len(indices_to_check) > 1:  # sort by iom
+                    iom_to_check = [
+                        (j, first.intersection_over_minimum(elements[j]))
+                        for j in indices_to_check
+                        if elements[j] is not None
+                    ]
+                    iom_to_check.sort(
+                        key=lambda x: x[1],
+                        reverse=True,
+                    )  # sort elements by iom, so we first check the greatest
+                    indices_to_check = [x[0] for x in iom_to_check]  # type:ignore
+            for j in indices_to_check:
+                if i != j and elements[j] is not None and elements[i] is not None:
+                    second = elements[j]
+                    intersection = first.intersection(
+                        second,
+                    )  # we know it does, but need the region
+                    first_inside_second = first.is_in(second)
+                    second_inside_first = second.is_in(first)
+
+                    if first_inside_second and not second_inside_first:
+                        elements[i] = None  # type:ignore
+                    elif second_inside_first and not first_inside_second:
+                        # delete second element
+                        elements[j] = None  # type:ignore
+                    elif intersection:
+                        iom = first.intersection_over_minimum(second)
+                        if iom < iom_to_merge:  # small
+                            separate(first, second)
+                            # The rectangle could become too small, which is a
+                            # good size to delete?
+                        else:  # big
+                            # merge
+                            if first.area > second.area:
+                                grow_region_to_match_region(first, second)
+                                elements[j] = None  # type:ignore
+                            else:
+                                grow_region_to_match_region(second, first)
+                                elements[i] = None  # type:ignore
+
+        elements = [e for e in elements if e is not None]
+        return elements
+
+    @staticmethod
+    def clean_type(elements: List[LayoutElement], type="Table") -> List[LayoutElement]:
```

Comment:
do this new functions added here needs to be in the class? It seems they can just be functions in this sub module and do not need to be part of the class. Especially consider long term reuse maybe it is preferred to have those functions being outside of the class definition?

Reply:
Probably `clean_type` could be outside, but I think `deduplicate_detected_elements` should remain here as only applies to detection models...do you have any particular suggestions on where we can place them?

Reply:
I am thinking because those operate on `LayoutElement` so they can be where those are defined. So in this case `unstructure_inference/inference/layoutelements.py`
The idea is that those functions really do not have any idea it is inside a model nor do they need to be. The most relevant context is how `LayoutElement` is defined. So it would seem they are better treated as tools applied to layout elements rather than tools bound to a model (even though right now we only have models using it). Models use the tools just like models depend on layout element's specific definitions. But models doesn't define those tools or layout elements' definitions.
With that said I can let it slide for this ticket and create a followup ticket to refactor these functions out.